src/core/public/application/application_service.tsx

```diff
@@ -282,8 +282,19 @@ export class ApplicationService {
       history: this.history!,
       getUrlForApp: (
         appId,
-        { path, absolute = false }: { path?: string; absolute?: boolean } = {}
+        {
+          path,
+          absolute = false,
+          deepLinkId,
+        }: { path?: string; absolute?: boolean; deepLinkId?: string } = {}
       ) => {
+        if (deepLinkId) {
+          const deepLinkPath = getAppDeepLinkPath(availableMounters, appId, deepLinkId);
+          if (deepLinkPath) {
+            path = appendAppPath(deepLinkPath, path);
+          }
+        }
```

Comment:
Implementation looks fine, will need to add unit tests in `src/core/public/application/application_service.test.ts` though.

Reply:
Hello @pgayvallet, 

Thank you for the prompt review. I pushed a new commit with unit tests. Could you take a second look?